File: app/src/test/java/com/hanstcs/movieapp/data/repository/MoviesRepositoryTest.java
package com.hanstcs.movieapp.data.repository;

import com.hanstcs.movieapp.data.Movies;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.MockitoAnnotations;

import java.util.ArrayList;
import java.util.List;

public class MoviesRepositoryTest {

    private static List<Movies> MOVIES;

    @Mock
    private MoviesRepository mMoviesRepository;

    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);

        MOVIES = new ArrayList<>();
        MOVIES.add(new Movies("1", "The lord of the rings", "2009-12-10"));
        MOVIES.add(new Movies("2", "The lord of the rings 2", "2012-12-10"));
    }

    @Test
    public void getMovies_willReturnListOfMovies() {
        Mockito.when(mMoviesRepository.getMovies()).thenReturn(MOVIES);
    }

    @Test
    public void getInstance_willReturnInstanceOfMoviesRepository() {
        mMoviesRepository = MoviesRepository.getInstance();
        Assert.assertSame(MoviesRepository.getInstance(), mMoviesRepository);
    }
}

File: app/src/test/java/com/hanstcs/movieapp/data/repository/MoviesRepositoryTest.kt
package com.hanstcs.movieapp.data.repository

import com.hanstcs.movieapp.data.Movies

import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.mockito.MockitoAnnotations

class MoviesRepositoryTest {

    private lateinit var MOVIES: List<Movies>

    private lateinit var moviesRepository: MoviesRepository

    @Before
    fun setUp() {
        MockitoAnnotations.initMocks(this)

        moviesRepository = MoviesRepository.getInstance()
        MOVIES = ArrayList()

        (MOVIES as ArrayList<Movies>).add(Movies("1", "The lord of the rings", "2009-12-10"))
        (MOVIES as ArrayList<Movies>).add(Movies("2", "The fallen and rise of the rome empire", "2017-10-25"))
        (MOVIES as ArrayList<Movies>).add(Movies("3", "Thor: Ragnarok", "2017-10-25"))
    }

    @Test
    fun getMovies_willReturnListOfMovies() {
        Assert.assertEquals(moviesRepository.getMovies(), MOVIES)
    }

    @Test
    fun getInstance_willReturnInstanceOfMoviesRepository() {
        moviesRepository = MoviesRepository.getInstance()
        Assert.assertSame(MoviesRepository.getInstance(), moviesRepository)
    }
}
